ibm/resource_ibm_is_subnet_public_gateway_attachment.go

```diff
@@ -0,0 +1,349 @@
+		Delete:   resourceIBMISSubnetPublicGatewayAttachmentDelete,
+		Exists:   resourceIBMISSubnetPublicGatewayAttachmentExists,
+		Importer: &schema.ResourceImporter{},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			isSubnetID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The subnet identifier",
+			},
+
+			isPublicGatewayID: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The unique identifier of network ACL",
+			},
+
+			isPublicGatewayName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Name of the Public gateway instance",
+			},
+
+			isPublicGatewayFloatingIP: {
+				Type:     schema.TypeMap,
+				Computed: true,
+			},
+
+			isPublicGatewayStatus: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Public gateway instance status",
+			},
+
+			isPublicGatewayResourceGroup: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Public gateway resource group info",
+			},
+
+			isPublicGatewayVPC: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Public gateway VPC info",
+			},
+
+			isPublicGatewayZone: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Public gateway zone info",
+			},
+
+			IsPublicGatewayResourceType: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The name of the resource",
+			},
+
+			isPublicGatewayCRN: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The crn of the resource",
+			},
+
+			ResourceGroupName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The resource group name in which resource is provisioned",
+			},
+		},
+	}
+}
+
+func resourceIBMISSubnetPublicGatewayAttachmentCreate(d *schema.ResourceData, meta interface{}) error {
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return err
+	}
+
+	subnet := d.Get(isSubnetID).(string)
+	publicGateway := d.Get(isPublicGatewayID).(string)
+
+	publicGatewayIdentity := &vpcv1.PublicGatewayIdentity{
+		ID: &publicGateway,
+	}
+
+	// Construct an instance of the SetSubnetPublicGatewayOptions
+	setSubnetPublicGatewayOptions := &vpcv1.SetSubnetPublicGatewayOptions{
+		ID:                    &subnet,
+		PublicGatewayIdentity: publicGatewayIdentity,
+	}
+
+	pg, response, err := sess.SetSubnetPublicGateway(setSubnetPublicGatewayOptions)
+
+	if err != nil {
+		log.Printf("[DEBUG] Error while attaching public gateway(%s) to subnet(%s) %s\n%s", publicGateway, subnet, err, response)
+		return fmt.Errorf("Error while attaching public gateway(%s) to subnet(%s) %s\n%s", publicGateway, subnet, err, response)
+	}
+	d.SetId(subnet)
```

Comment:
ID could be in the form of <subnet_id>/<publicgateway_id>

Reply:
we dont need public gateway id as only one attachment is possible